cluster-autoscaler/cloudprovider/azure/azure_config.go

```diff
@@ -154,177 +116,126 @@ func BuildAzureConfig(configReader io.Reader) (*Config, error) {
-			cfg.VmssVmsCacheJitter, err = strconv.Atoi(vmssVmsCacheJitter)
-			if err != nil {
-				return nil, fmt.Errorf("failed to parse AZURE_VMSS_VMS_CACHE_JITTER %q: %v", vmssVmsCacheJitter, err)
-			}
-		}
-
-		if threshold := os.Getenv("AZURE_MAX_DEPLOYMENT_COUNT"); threshold != "" {
-			cfg.MaxDeploymentsCount, err = strconv.ParseInt(threshold, 10, 0)
-			if err != nil {
-				return nil, fmt.Errorf("failed to parse AZURE_MAX_DEPLOYMENT_COUNT %q: %v", threshold, err)
-			}
-		}
+	}
 
-		if enableBackoff := os.Getenv("ENABLE_BACKOFF"); enableBackoff != "" {
-			cfg.CloudProviderBackoff, err = strconv.ParseBool(enableBackoff)
-			if err != nil {
-				return nil, fmt.Errorf("failed to parse ENABLE_BACKOFF %q: %v", enableBackoff, err)
-			}
+	// Each of these environment variables, if provided, will override what's in the config file.
+	// Note that this "retrieval from env" does not exist in cloud-provider-azure library (at the time of this comment).
+	if _, err = assignFromEnvIfExists(&cfg.Location, "LOCATION"); err != nil {
+		return nil, err
+	}
+	if _, err = assignFromEnvIfExists(&cfg.ResourceGroup, "ARM_RESOURCE_GROUP"); err != nil {
+		return nil, err
+	}
+	if _, err = assignFromEnvIfExists(&cfg.TenantID, "ARM_TENANT_ID"); err != nil {
+		return nil, err
+	}
+	if _, err = assignFromEnvIfExists(&cfg.TenantID, "AZURE_TENANT_ID"); err != nil {
+		return nil, err
+	}
+	if _, err = assignFromEnvIfExists(&cfg.AADClientID, "ARM_CLIENT_ID"); err != nil {
+		return nil, err
+	}
+	if _, err = assignFromEnvIfExists(&cfg.AADClientID, "AZURE_CLIENT_ID"); err != nil {
+		return nil, err
+	}
+	if _, err = assignFromEnvIfExists(&cfg.AADFederatedTokenFile, "AZURE_FEDERATED_TOKEN_FILE"); err != nil {
+		return nil, err
+	}
+	if _, err = assignFromEnvIfExists(&cfg.AADClientSecret, "ARM_CLIENT_SECRET"); err != nil {
+		return nil, err
+	}
+	if _, err = assignFromEnvIfExists(&cfg.VMType, "ARM_VM_TYPE"); err != nil {
+		return nil, err
+	}
+	if _, err = assignFromEnvIfExists(&cfg.AADClientCertPath, "ARM_CLIENT_CERT_PATH"); err != nil {
+		return nil, err
+	}
+	if _, err = assignFromEnvIfExists(&cfg.AADClientCertPassword, "ARM_CLIENT_CERT_PASSWORD"); err != nil {
+		return nil, err
+	}
+	if _, err = assignFromEnvIfExists(&cfg.Deployment, "ARM_DEPLOYMENT"); err != nil {
+		return nil, err
+	}
+	if _, err = assignFromEnvIfExists(&cfg.SubscriptionID, "ARM_SUBSCRIPTION_ID"); err != nil {
+		return nil, err
+	}
+	if _, err = assignBoolFromEnvIfExists(&cfg.UseManagedIdentityExtension, "ARM_USE_MANAGED_IDENTITY_EXTENSION"); err != nil {
+		return nil, err
+	}
+	if _, err = assignBoolFromEnvIfExists(&cfg.UseFederatedWorkloadIdentityExtension, "ARM_USE_WORKLOAD_IDENTITY_EXTENSION"); err != nil {
+		return nil, err
+	}
+	if _, err = assignFromEnvIfExists(&cfg.UserAssignedIdentityID, "ARM_USER_ASSIGNED_IDENTITY_ID"); err != nil {
+		return nil, err
+	}
+	if _, err = assignIntFromEnvIfExists(&cfg.VmssCacheTTLInSeconds, "AZURE_VMSS_CACHE_TTL"); err != nil {
```

Comment:
This is one of the fields that have a difference in naming (`cloud-provider-azure` version has "InSeconds"). Although we keep the environment variables the same to not introduce breaking changes. Not that environment variable name needs to be changed either, given the minimal change.